metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -692,6 +702,136 @@ def get_lineage_if_enabled(
+                column_profile.min = column_params[self.source_config.profiling.min]
+            if self.source_config.profiling.max in column_params:
+                column_profile.max = column_params[self.source_config.profiling.max]
+            if self.source_config.profiling.mean in column_params:
+                column_profile.mean = column_params[self.source_config.profiling.mean]
+            if self.source_config.profiling.median in column_params:
+                column_profile.median = column_params[
+                    self.source_config.profiling.median
+                ]
+            if self.source_config.profiling.stdev in column_params:
+                column_profile.stdev = column_params[self.source_config.profiling.stdev]
+
+            dataset_profile.fieldProfiles.append(column_profile)
+
+        if partition_spec:
+            # inject partition level stats
+            dataset_profile.partitionSpec = PartitionSpecClass(
+                partition=partition_spec,
+                type=PartitionTypeClass.PARTITION,
+            )
+
+        mcp = MetadataChangeProposalWrapper(
+            entityType="dataset",
+            entityUrn=mce.proposedSnapshot.urn,
+            changeType=ChangeTypeClass.UPSERT,
+            aspectName="datasetProfile",
+            aspect=dataset_profile,
+        )
+        return mcp
+
+    def get_profile_if_enabled(
+        self, mce: MetadataChangeEventClass, database_name: str, table_name: str
+    ) -> List[Optional[MetadataChangeProposalWrapper]]:
+        if self.source_config.profiling.enabled:
+            if self.source_config.profiling.partitioned:
+                # for cross-account ingestion
+                if self.source_config.catalog_id:
+                    response = self.glue_client.get_partitions(
+                        DatabaseName=database_name,
+                        Name=table_name,
+                        CatalogId=self.source_config.catalog_id,
+                    )
+                else:
+                    response = self.glue_client.get_partitions(
+                        DatabaseName=database_name, Name=table_name
+                    )
+
+                partitions = response["partitions"]
+
+                mcps = []
+                for p in partitions:
+                    table_stats = p["Parameters"]
+                    column_stats = p["Table"]["StorageDescriptor"]["Columns"]
+                    # only support single partition key
+                    partition_spec = str(
+                        {self.source_config.profiling.partition_key: p["Values"][0]}
+                    )
+                    mcps.append(
+                        self._create_profile_mcp(
+                            mce, table_stats, column_stats, partition_spec
+                        )
+                    )
+                return mcps
+            else:
+                # for cross-account ingestion
+                if self.source_config.catalog_id:
```

Comment:
same here as above, this check is not necessary.

Reply:
See same reply as above